lib/ansible/modules/system/user.py

```diff
@@ -1813,6 +1866,24 @@ def modify_user_usermod(self):
 
         return (rc, out, err)
 
+    def user_info(self):
+        info = super(SunOS, self).user_info()
+        if info:
+            info += self._user_attr_info()
+        return info
+
+    def _user_attr_info(self):
+        info = [''] * 3
+        with open(self.USER_ATTR, 'rb') as file_handler:
+            for line in file_handler:
+                line = to_native(line.strip())
+                if self.name in line:
+                    tmp = dict(map(lambda x: x.split('='), line.split('::::')[1].split(';')))
```

Comment:
Rather than use a `lamdba` here, it would be better to use a generator expression for clarity
```suggestion
                    lines = line.split('::::')[1].split(';')
                    tmp = dict(x.split('=') for x in lines)
                    
```

Reply:
Updated as per suggestion, although I also did some change with user checking (previously it could have a false match)